### ISSUE TITLE
Delete profile concerns

### ISSUE THREAD

**Issue by cmuld3r** [SandersForPresident/RideWithBernie#19]
How should we handle deleting user profiles? We don't want bad data and people getting contacted when they've already found a ride.

Do we need an easier way to delete a profile? Maybe a text message that auto-deletes, or just includes a shortlink to the profile page which scrolls to the delete button?

Also, do we already keep, or do we need to keep every record after 'deleting' for statistical purposes?

**Comment by buddhistsforbernie** [comment]
It seems intuitive that when a driver contacts a passenger, that passenger
gets deleted, or at least marked accounted for?

As for drivers, I think some drivers might be bussing people all day to
some events (e.g. PRIMARIES!), so maybe just giving them an option to mark
themselves "full"?

**Comment by kyleschmolze** [comment]
Do you think the current way to delete a profile is too hidden? It may be, but maybe we should wait to hear from real users before we're sure? Also, a good middle-ground option between auto-deleting and not might be: After a user gets contacted by a driver/passenger, we send a follow up text that says "If you are all set, please make sure to delete your profile here: [LINK]"

**Comment by cmuld3r** [comment]
@buddhistsforbernie i don't think we should delete a passenger when a driver contacts them... as they may not actually coordinate with each other... and another driver may want to contact the passenger. We have it setup where you can contact multiple people and wait for a response.

**Comment by cmuld3r** [comment]
i do think it's a bit hidden, as we never send people directly to the profile. the current profile page link takes them back to the search page. I like the follow-up text idea to give them a link directly to their profile.

**Comment by clarissa1287** [comment]
I like the text idea as well, but what if they want to delete their profile before the ride has gone through? How would they delete their profile before they get the text?

**Comment by kyleschmolze** [comment]
Right now, after you create a profile, if you go back to edit it, there's a delete button.

**Comment by clarissa1287** [comment]
That's awesome! Thanks @kyletns!

**Comment by cmuld3r** [comment]
hey @kyletns, with deleting profile records we could be losing stats about people who are signing up and contacting other people, right? think this is an issue? if so, is there a way to have stats on people who end up deleting their profile even though we're deleting records?

**Comment by cmuld3r** [comment]
if we can't currently, what do you think about a delete flag or having a journal table with all records?

**Comment by kyleschmolze** [comment]
I would opt for the "delete flag" option, @cmuld3r. Maybe check out https://github.com/rubysherpas/paranoia, which I've used to success before. Also, it may be smart to add a model for Sms or TextMessage or something, so we can keep a history of everything we're sending out.

**Comment by cmuld3r** [comment]
what would be the rationale for keep the texts we're sending? in case we want to display in the app at a later date?

**Comment by cmuld3r** [comment]
thanks, i'll check out paranoia :)

**Comment by kyleschmolze** [comment]
No just for our own purposes, for debugging.
